public/src/components/utilities/QrCodePage.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState } from 'react';
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  FormControl,
+  makeStyles,
+  Paper,
+  TextField,
+  Theme,
+  Typography,
+} from '@material-ui/core';
+import QRCode from 'react-qr-code';
+
+const useStyles = makeStyles(({ spacing }: Theme) => ({
+  container: {
+    height: '100%',
+    overflowY: 'auto',
+    padding: spacing(2),
+    display: 'grid',
+    gridTemplateColumns: 'repeat(5, 1fr)',
+    gridTemplateRows: 'repeat(2, 1fr)',
+  },
+  subContainer: {
+    margin: '0 auto',
+    display: 'flex',
+    flexDirection: 'column',
+    width: '650px',
+    padding: spacing(2),
+    gridArea: ' 1 / 2 / 3 / 5',
+    alignSelf: 'start',
+  },
+  heading: {
+    fontSize: '1.5rem',
+    marginBottom: spacing(2),
+  },
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    '& > *:not(:last-child)': {
+      marginBottom: spacing(2),
+    },
+  },
+  codeContainer: {
+    minWidth: '90%',
+    marginTop: spacing(4),
+    marginBottom: spacing(4),
+    padding: spacing(4),
+    display: 'flex',
+    justifyContent: 'center',
+  },
+}));
+
+const userAdviceStyles = makeStyles({
+  card: {
+    alignSelf: 'start',
+    justifySelf: 'center',
+    maxWidth: '360px',
+  },
+  paragraph: {
+    marginBottom: '1rem',
+  },
+});
+
+function UserAdvice() {
+  const classes = userAdviceStyles();
+  return (
+    <Card className={classes.card}>
+      <CardContent>
+        <Typography className={classes.paragraph}>
+          When using the QR code to track a reader by linking it to a URL you must ensure that the
+          reader&apos;s CMP choice is respected either by reference to the reader choice logged with
+          Sourcepoint, or by reference to the correct pageview table.
+        </Typography>
+        <Typography>For further guidance contact the T&C team or the Data Privacy team.</Typography>
+      </CardContent>
+    </Card>
+  );
+}
+
+export default function QrCodePage(): JSX.Element {
+  const [url, setUrl] = useState('');
+  const [fileName, setFileName] = useState('');
+  const [codeSize, setCodeSize] = useState(256);
+  const classes = useStyles();
+
+  function onSvgDownload() {
+    const svg = document.querySelector<SVGElement>('#QRCode');
```

Comment:
perhaps not necessary, but would a `useRef` work here?

Reply:
I tried, but the `QrCode` component doesn't accept a ref.